### PR TITLE
Adding unit tests to #278

### PR DESCRIPTION
**Pull request by jmprieur**
Adding unit tests to #278 and reverting the UI package upgrade as 3.1.1 has a bug

## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ x] Other... unit tests
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -114,11 +115,8 @@
 /2-WebApp-graph-user/2-4-Sovereign-Call-MSGraph/bin/Debug/netcoreapp2.2
 /2-WebApp-graph-user/2-4-Sovereign-Call-MSGraph/obj
 /2-WebApp-graph-user/2-4-Sovereign-Call-MSGraph/bin/Release/netcoreapp2.2
-/Microsoft.Identity.Web.Test/bin/Release/netcoreapp2.2
+/Microsoft.Identity.Web.Test/bin
 /Microsoft.Identity.Web.Test/obj
 /4-WebApp-your-API/4-2-B2C/.vs
 /4-WebApp-your-API/4-2-B2C/Client/obj
 /4-WebApp-your-API/4-2-B2C/TodoListService/obj
-/2-WebApp-graph-user/2-3-Multi-Tenant/.vs/WebApp-OpenIDConnect-DotNet
```
Comment (jmprieur) [inline .gitignore]: Was already present in the .gitignore

File: 2-WebApp-graph-user/2-1-Call-MSGraph/AspnetCoreWebApp-calls-Microsoft-Graph.sln [hunk 2]
```diff
@@ -7,6 +7,13 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "WebApp-OpenIDConnect-DotNet
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Identity.Web", "..\..\Microsoft.Identity.Web\Microsoft.Identity.Web.csproj", "{E0CEF26A-6CE6-4505-851B-6580D5564752}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{CD0BB564-6C1E-4FCE-B9AB-7C637FDEE569}"
+	ProjectSection(SolutionItems) = preProject
+		.editorconfig = .editorconfig
+	EndProjectSection
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Identity.Web.Test", "..\..\Microsoft.Identity.Web.Test\Microsoft.Identity.Web.Test.csproj", "{8CCEAE2A-BDF6-470C-B6DE-7FC81A74DBD7}"
```
Comment (jmprieur) [inline 2-WebApp-graph-user/2-1-Call-MSGraph/AspnetCoreWebApp-calls-Microsoft-Graph.sln]: Adding the unit tests assembly to the solution

File: 2-WebApp-graph-user/2-1-Call-MSGraph/WebApp-OpenIDConnect-DotNet.csproj [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Authentication.AzureAD.UI" Version="3.0.0" />
-    <PackageReference Include="Microsoft.Graph" Version="1.14.0" />
+    <PackageReference Include="Microsoft.Graph" Version="1.21.0" />
```
Comment (jmprieur) [inline 2-WebApp-graph-user/2-1-Call-MSGraph/WebApp-OpenIDConnect-DotNet.csproj]: Updating to the latest Graph SDK

File: Microsoft.Identity.Web/Microsoft.Identity.Web.csproj [hunk 4]
```diff
@@ -53,8 +53,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Authentication.AzureAD.UI" Version="3.1.1" />
-    <PackageReference Include="Microsoft.AspNetCore.Authentication.AzureADB2C.UI" Version="3.1.1" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.AzureAD.UI" Version="3.0.0" />
```
Comment (jmprieur) [inline Microsoft.Identity.Web/Microsoft.Identity.Web.csproj]: AzureADB2C.UI 3.1.1 has an issue.

File: Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -53,17 +53,11 @@ public static class WebApiServiceCollectionExtensions
                 configuration.Bind(configSectionName, options);
 
                 // This is an Microsoft identity platform Web API
-                var authority = options.Authority.Trim().TrimEnd('/');
-                if (!authority.EndsWith("v2.0"))
-                    authority += "/v2.0";
-                options.Authority = authority;
+                EnsureAuthorityIsV2_0(options);
```
Comment (jmprieur) [inline Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: Refactoring for testability

File: Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs [hunk 6]
```diff
@@ -53,17 +53,11 @@ public static class WebApiServiceCollectionExtensions
...
 
-                // The valid audience could be given as Client Id or as Uri. If it does not start with 'api://', this variant is added to the list of valid audiences.
-                var validAudiences = new List<string> { options.Audience };
-                if (!options.Audience.StartsWith("api://", StringComparison.OrdinalIgnoreCase))
-                    validAudiences.Add($"api://{options.Audience}");
-
-                options.TokenValidationParameters.ValidAudiences = validAudiences;
+                // The valid audience could be given as Client Id or as Uri. 
+                // If it does not start with 'api://', this variant is added to the list of valid audiences.
+                EnsureValidAudiencesContainsApiGuidIfGuidProvided(options);
```
Comment (jmprieur) [inline Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs]: Refactoring for testability

File: Microsoft.Identity.Web.Test/WebApiServiceCollectionExtensionsTests.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace Microsoft.Identity.Web.Test
+{
+    public class WebApiServiceCollectionExtensionsTests
+    {
+        [Fact]
+        public void TestAuthority()
+        {
+            // Arrange
+            JwtBearerOptions options = new JwtBearerOptions();
```
Comment (jennyf19) [inline Microsoft.Identity.Web.Test/WebApiServiceCollectionExtensionsTests.cs]: do we need to cover sovereign clouds as well? maybe as a work item?

File: Microsoft.Identity.Web.Test/WebApiServiceCollectionExtensionsTests.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using System;
```
Comment (jennyf19) [inline Microsoft.Identity.Web.Test/WebApiServiceCollectionExtensionsTests.cs]: license info?

File: Microsoft.Identity.Web.Test/WebApiServiceCollectionExtensionsTests.cs (hunk 7 above)
Comment (jmprieur) [reply Microsoft.Identity.Web.Test/WebApiServiceCollectionExtensionsTests.cs]: I'm not sure what this changes (the domain would be different). But I added one test.

File: Microsoft.Identity.Web.Test/WebApiServiceCollectionExtensionsTests.cs (hunk 8 above)
Comment (jmprieur) [reply Microsoft.Identity.Web.Test/WebApiServiceCollectionExtensionsTests.cs]: Added. Thanks!

**Pull request merged**